src/js/personalization/va-profile-beta/utils/tests/address.unit.spec.js

```diff
@@ -0,0 +1,118 @@
+import * as addressUtils from '../address';
+import { expect } from 'chai';
+
+const domestic = {
+  type: 'DOMESTIC',
+  countryName: 'USA',
+  addressOne: 'First',
+  addressTwo: 'Second',
+  addressThree: 'Third',
+  city: 'City',
+  stateCode: 'KY',
+  zipCode: '12345'
+};
+
+const international = {
+  type: 'INTERNATIONAL',
+  countryName: 'Canada',
+  addressOne: 'First',
+  addressTwo: 'Second',
+  addressThree: 'Third',
+  city: 'Montreal'
+};
+
+const military = {
+  type: 'MILITARY',
+  addressOne: 'First',
+  addressTwo: 'Second',
+  addressThree: 'Third',
+  militaryPostOfficeTypeCode: 'APO',
+  militaryStateCode: 'AA',
+  zipCode: '12345'
+};
+
+describe('formatAddress', () => {
+
+  it('formats domestic addresses', () => {
+    const expectedResult = {
+      street: 'First, Second Third',
+      cityStateZip: 'City, Kentucky 12345',
+      country: ''
+    };
+
+    expect(addressUtils.formatAddress(domestic)).to.deep.equal(expectedResult);
+  });
+
+  it('formats military addresses', () => {
+    const expectedResult = {
+      street: 'First, Second Third',
+      cityStateZip: 'APO, AA 12345',
+      country: ''
+    };
+
+    expect(addressUtils.formatAddress(military)).to.deep.equal(expectedResult);
+  });
+
+  it('formats international addresses', () => {
+    const expectedResult = {
+      street: 'First, Second Third',
+      cityStateZip: 'Montreal',
+      country: 'Canada'
+    };
+
+    expect(addressUtils.formatAddress(international)).to.deep.equal(expectedResult);
+  });
+
+});
+
+describe('getStateName', () => {
+
+  it('gets the full name of a state from its abbreviation', () => {
+    expect(addressUtils.getStateName('KY')).to.equal('Kentucky');
+    expect(addressUtils.getStateName('ky')).to.equal('Kentucky');
+  });
+
+});
+
+describe('isEmptyAddress', () => {
+
+  it('detects an empty address', () => {
+    expect(addressUtils.isEmptyAddress({})).to.equal(true);
+    expect(addressUtils.isEmptyAddress({ type: 'domestic', countryName: 'USA' })).to.equal(true);
+  });
+
+  it('detects a non-empty address', () => {
+    expect(addressUtils.isEmptyAddress(domestic)).to.equal(false);
+  });
+
+});
+
+describe('consolidateAddress', () => {
+  it('converts a military address into a standard address format', () => {
+    const expectedResult = {
+      type: 'MILITARY',
+      countryName: 'USA',
+      addressOne: 'First',
+      addressTwo: 'Second',
+      addressThree: 'Third',
+      city: 'APO',
+      stateCode: 'AA',
+      zipCode: '12345'
```

Comment:
Do military addresses have ZIP codes?

Reply:
They do! But I'll update my example to match the [API's](https://github.com/department-of-veterans-affairs/vets-api/blob/1efd2c206859b1a261e762a50cdb44dc8b66462d/spec/factories/pciu_address.rb#L34)